03_Modules/Certificates/src/module/api.ts

```diff
@@ -228,6 +236,121 @@ export class CertificatesModuleApi
+      : 2048;
+    certificateFromReq.organizationName = certRequest.organizationName;
+    certificateFromReq.commonName = certRequest.commonName;
+    if (certRequest.validBefore) {
+      certificateFromReq.validBefore = certRequest.validBefore;
+    } else {
+      const defaultValidityDate: Date = new Date();
+      defaultValidityDate.setFullYear(defaultValidityDate.getFullYear() + 1);
+      certificateFromReq.validBefore = defaultValidityDate.toISOString();
+    }
+
+    let responseBody: Certificate[];
+    let rootCAPem;
+    if (certRequest.selfSigned) {
+      // Generate self-signed root CA certificate
+      const [rootCertificatePem, rootPrivateKeyPem] =
+        this._generateSelfSignedRootCertificate(certificateFromReq);
+      certificateFromReq = await this._storeCertificateAndKey(
+        certificateFromReq,
+        rootCertificatePem,
+        rootPrivateKeyPem,
+        PemType.Root,
+        certRequest.filePath,
+      );
+
+      // Generate sub CA certificate
+      let subCertificate: Certificate = new Certificate();
+      subCertificate.certificateType = certificateFromReq.certificateType;
+      subCertificate.serialNumber = this._generateSerialNumber();
+      subCertificate.keyLength = certificateFromReq.keyLength;
+      subCertificate.organizationName = certificateFromReq.organizationName;
+      subCertificate.commonName = certRequest.commonName + ' Sub CA';
+      subCertificate.validBefore = certificateFromReq.validBefore;
+      subCertificate.signedBy = certificateFromReq.id;
+      const [subCertificatePem, subPrivateKeyPem] = this._generateCertificate(
+        rootPrivateKeyPem,
+        rootCertificatePem,
+      );
+      subCertificate = await this._storeCertificateAndKey(
+        subCertificate,
+        subCertificatePem,
+        subPrivateKeyPem,
+        PemType.SubCA,
+        certRequest.filePath,
+      );
+
+      responseBody = [subCertificate, certificateFromReq];
+      rootCAPem = rootCertificatePem;
+    } else {
+      // Get root certificate from external CA
+      const externalRootCAPem =
+        await this._module.certificateAuthorityService.getRootCACertificateFromExternalCA();
+
+      // Generate sub CA certificate and private key
+      const [certificatePem, privateKeyPem] =
+        await this._generateSubCACertificateSignedByCAServer(
+          certificateFromReq,
+        );
+      certificateFromReq = await this._storeCertificateAndKey(
+        certificateFromReq,
+        certificatePem,
+        privateKeyPem,
+        PemType.SubCA,
+        certRequest.filePath,
+      );
+
+      responseBody = [certificateFromReq];
+      rootCAPem = externalRootCAPem;
+    }
+
+    // Send InstallCertificateRequest to the charger
+    const confirmation: IMessageConfirmation = await this.installCertificate(
+      certRequest.stationId,
+      certRequest.tenantId,
+      {
+        certificateType: certRequest.certificateType,
+        certificate: rootCAPem,
```

Comment:
Here I change to send only the root CA which is different from what was decided in our previous discussion, i.e., send a sub CA and root CA chain in the installCertificate request. Since during testing with everest, I found they expect a single certificate. 
<img width="1100" alt="Screenshot 2024-05-07 at 5 34 34 PM" src="https://github.com/citrineos/citrineos-core/assets/21100540/ed483489-78ea-421b-a4f2-99e7df56dcf3">
I think this also makes sense. Because in A02 update cert case, it expects for a certificate chain in the CertificateSignedRequest (leaf and the necessary sub CA certificates). And we might don't need to send the sub CA in install request.